egui/src/widgets/spinner.rs

```diff
@@ -0,0 +1,65 @@
+use epaint::{emath::lerp, vec2, Pos2, Shape, Stroke};
+
+use crate::{Response, Sense, Ui, Widget};
+
+/// A spinner widget used to indicate loading.
+#[must_use = "You should put this widget in an ui with `ui.add(widget);`"]
+pub struct Spinner {
+    enabled: bool,
+    /// Uses the style's `interact_size` if `None`.
+    size: Option<f32>,
+}
+
+impl Spinner {
+    /// A new spinner that is shown if `enabled` is true.
+    /// Uses the style's `interact_size` unless changed.
+    /// A disabled spinner still takes up it's usual space in order to prevent inconsistent
+    /// alignment.
+    pub fn new(enabled: bool) -> Self {
```

Comment:
No need for this! I just pushed https://github.com/emilk/egui/commit/4ef21380a25eb842a0f4aff66825f2b9faf1ecc5 where I added a method `Ui::add_visible` so you can do `ui.add_visible(is_loading, egui::Spinner::new());`